corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -676,9 +768,83 @@ hqDefine("cloudcare/js/formplayer/menus/views", [
+            if (configButton.length) {
+                const caseListConfigViewPopover = new bootstrap.Popover(configButton[0], {
+                    html: true,
+                    sanitize: false,
+                    content: function () {
+                        self.caseListConfigView = new CaseListConfigView({
+                            model: self.columnConfigModel,
+                        });
+                        const container = document.createElement('div');
+                        self.caseListConfigView.setElement(container);
+                        self.caseListConfigView.render();
+
+                        self.listenTo(self.caseListConfigView, 'save', function () {
+                            caseListConfigViewPopover.dispose();
+                            // save map
+                            const initialixedMapEl = self.$('#module-case-list-map')[0];
+
+                            self.render();
+
+                            // Replace the map element with the initial map element
+                            const newMapEl = self.$('#module-case-list-map')[0];
+                            newMapEl.parentNode.replaceChild(initialixedMapEl, newMapEl);
+                        });
+
+                        return container;
+                    },
+                    placement: 'auto',
+                    trigger: 'click',
+                });
+
+                document.addEventListener('click', function (event) {
+                    if ($(event.target).closest('.popover').length) {
+                        if (!$(event.target).hasClass('js-action') &&
+                                !$(event.target).closest('.js-action').length) {
+                            event.stopPropagation();
+                        }
+                    } else {
+                        caseListConfigViewPopover.hide();
+                    }
+                }, true);
+
+                configButton[0].addEventListener('shown.bs.popover', function () {
+                    const popover = document.querySelector('.popover');
+                    if (popover) {
+                        const firstCheckbox = popover.querySelector('.column-checkbox');
+                        if (firstCheckbox) {
+                            firstCheckbox.focus();
+                        }
+                    }
+                });
+            }
+        },
+
+        getConfigStorageId: function (user) {
+            const urlObject = formplayerUtils.currentUrlToObject();
+            const selectionsWithoutUuid = urlObject.selections.map(function (s) {
+                if (s.match('^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$')) {
+                    return 'uuid';
+                } else {
+                    return s;
+                }
+            });
+            const configStorageId = `${urlObject.appId}:${JSON.stringify(selectionsWithoutUuid)}:${user.username}`;
```

Comment:
is there a need to include the user's username in the configStorageId? Is this so state persists unique to whoever you "sign in as"? If so, I don't think that's necessary. Partly because I saw the discussion about if it's safe to include username in the key and the discussion seems inconclusive.